pkg/model/crd.go

```diff
@@ -448,6 +466,24 @@ func (r *CRD) GoCodeSetInput(
 	out := "\n"
 	indent := strings.Repeat("\t", indentLevel)
 
+	// During initial creation of aws resource, return NotFoundError if all the required Status fields are missing for ReadOne call.
+	// Status fields will be populated after object creation
+	if OpTypeGet == opType {
+		var requiredKoStatusFields = r.RequiredStatusFieldsForReadOneInput()
+		if len(requiredKoStatusFields) > 0 {
+			allRequiredKoStatusFieldsMissingComment := "// If all the requiredKoStatusFields are missing, AWS resource is not created yet. Return NotFound exception\n"
```

Comment:
For this kind of thing where there is a long comment and an if condition on the operation type, I'd prefer to put this kind of thing into the template and not the Go code generator.

In other words, here:

https://github.com/aws/aws-controllers-k8s/blob/1cabf4d5a2a159a68ffe7f44d8a4965fe3034d47/templates/pkg/crd_sdk.go.tpl#L39

You would not need to have an `if` conditional because you're already in the "ReadOne" input shape block.

You could add this into the template:

```go
// If all the required fields in the input shape are missing, AWS
// resource is not created yet. Return NotFound here to indicate
// to callers that the resource isn't yet created.
if rm.requiredFieldsMissingFromReadOneInput(r) {
    return nil, ackerr.NotFound
}
```

and then later, in this section:

https://github.com/aws/aws-controllers-k8s/blob/1cabf4d5a2a159a68ffe7f44d8a4965fe3034d47/templates/pkg/crd_sdk.go.tpl#L103-L113

Add this:

```go
func (rm *resourceManager) requiredFieldsMissingFromReadOneInput(
    r *resource,
) bool {
{{ GoCodeRequiredFieldsFromReadOneInput "r" }}
}
```

And then all you need to do is implement a separate `CRD.GoCodeRequiredFieldsFromReadOneInput` method.